Sources/Datadog/Core/PerformancePreset.swift

```diff
@@ -68,56 +68,80 @@ internal struct PerformancePreset: Equatable, StoragePerformancePreset, UploadPe
     let minUploadDelay: TimeInterval
     let maxUploadDelay: TimeInterval
     let uploadDelayChangeRate: Double
+}
 
-    // MARK: - Predefined presets
-
-    /// Default performance preset.
-    static let `default` = lowRuntimeImpact
-
-    /// Performance preset optimized for low runtime impact.
-    /// Minimalizes number of data requests send to the server.
-    static let lowRuntimeImpact = PerformancePreset(
-        // persistence
-        maxFileSize: 4 * 1_024 * 1_024, // 4MB
-        maxDirectorySize: 512 * 1_024 * 1_024, // 512 MB
-        maxFileAgeForWrite: 4.75,
-        minFileAgeForRead: 4.75 + 0.5, // `maxFileAgeForWrite` + 0.5s margin
-        maxFileAgeForRead: 18 * 60 * 60, // 18h
-        maxObjectsInFile: 500,
-        maxObjectSize: 256 * 1_024, // 256KB
+internal extension PerformancePreset {
+    init(
+        batchSize: Datadog.Configuration.BatchSize,
+        uploadFrequency: Datadog.Configuration.UploadFrequency,
+        bundleType: BundleType
+    ) {
+        let meanFileAgeInSeconds: TimeInterval = {
+            switch (bundleType, batchSize) {
+            case (.iOSApp, .small): return 5
```

Comment:
Out of curiosity, what's the difference between iOSApp and iOSAppExtension ?

Reply:
this is due to GH issue number 52
(i'm not giving the actual refererence to it on purpose, no need to alert it i guess)

Reply:
[iOS App Extensions](https://developer.apple.com/app-extensions/) are the way to add functionality beyond the app so they can interact with other apps or the system (e.g. home screen widgets). 

We differentiate `PerformancePreset` in app extension, because its process is usually designed to work only for the **very short amount of time**, when the user interacts with it, e.g. chooses the Slack channel and message to upload photo directly from the iOS photo library through Slack Extension (w/o launching the Slack app).

Such `PerformancePreset` for app extension stores and uploads data much mor frequently + its initial upload is scheduled almost right away when the app extension starts. This is to make sure the SDK has enough chance to upload all data before the extension is closed (which usually happens in few seconds after it was started).